File: Java/1251-1300/1287. Element Appearing More Than 25% In Sorted Array.java
// https://leetcode.com/problems/element-appearing-more-than-25-in-sorted-array/

class Solution {
    public int findSpecialInteger(int[] arr) {
        
        int size = arr.length;
        int qtr = size / 4, count = 1;

        int result = arr[0];

        for(int index=1; index < size; index++){
            // Update count
            if(result == arr[index]) count++;
            else count = 1;

            if(count > qtr) return result;

            result = arr[index];
        }
        
        return result;
    }
}

File: Python/1251-1300/1287. Element Appearing More Than 25% In Sorted Array.py
# https://leetcode.com/problems/element-appearing-more-than-25-in-sorted-array/

class Solution:
    def findSpecialInteger(self, arr: List[int]) -> int:
        size = len(arr)
        qtr, count = size // 4, 1

        result = arr[0]

        for index in range (1, size):
            # Update count
            if result == arr[index]:
                count += 1
            else:
                count = 1

            if(count > qtr):
                return result

            result = arr[index]
        
        return result
